packages/modules/viewer-react/src/types.ts

```diff
@@ -131,21 +141,53 @@ export interface ViewerInitializerParams extends ViewerIModelAppOptions {
   /** array of iTwin.js Extensions */
   extensions?: ExtensionProvider[];
 }
-
-export interface ConnectedViewerProps {
-  iTwinId: string;
+export type RequiredViewerProps = XOR<
+  XOR<ConnectedViewerProps, FileViewerProps>,
+  BlankViewerProps
+>;
+export type ViewerProps = RequiredViewerProps & ViewerCommonProps;
+export type ViewerLoaderProps = RequiredViewerProps & LoaderProps;
+export type ConnectedViewerProps = {
+  iTwinId: GuidString;
   iModelId: string;
   changeSetId?: string;
-}
-
-export interface FileViewerProps {
+};
+export type FileViewerProps = {
   /** Path to local snapshot or briefcase */
   filePath: string;
-}
+};
+
+export type BlankViewerProps = XOR<
+  XOR<
+    {
+      /** @deprecated specify location and extents instead. */
+      blankConnection: PropRequired<BlankConnectionProps, "iTwinId">;
+      blankConnectionViewState?: BlankConnectionViewState;
+    },
+    {
+      /** @deprecated specify location and extents instead. */
```

Comment:
wonder if we need this one at all; it might be an unecessary step in between the deprecated api and where we're pushing the api fwd to; might as well have consumers just update to the preferred api.